File: authentication-server/src/main/java/com/andrew/authenticationserver/entity/auth/Role.java
package com.andrew.authenticationserver.entity.auth;


import lombok.Data;

import javax.persistence.*;
import java.util.List;

@Entity
@Table(name = "role")
@Data
public class Role {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(name = "role")
    private String role;
    @ManyToMany(mappedBy = "roles")
    private List<User> users;
}

File: auth-server/src/main/java/com/andrew/authserver/entity/auth/Role.kt
package com.andrew.authserver.entity.auth

import jakarta.persistence.*
import lombok.Data

@Entity
@Table(name = "role")
@Data
class Role {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private val id: Long = 1;
    @Column(name = "role")
    private var role: String? = null;
    @ManyToMany(mappedBy = "roles")
    private var users: List<User>? = null;

    fun getRole(): String? {
        return role;
    }

    fun setRole(role: String) {
        this.role = role;
    }

}